### PR TITLE
Migrate to Microsoft.IdentityModel.JsonWebTokens

### PR DESCRIPTION
**Pull request by iamcarbon**
This PR:

- Enables C#12
- Updates the library to use the modernized Microsoft.IdentityModel.JsonWebTokens library
- Updates the demos to use .net8.0 (the main library continues to target net6.0)
- Fixes various dotnet format violations
- Updates xunit to the latest version
- Updates System.Formats.Cbor to 8.0.0
- Replaces DataHelper.Concat with collection expressions

### CODE REVIEW

**Comment by iamcarbon** [comment]
@abergs Ready for review.

File: Src/Fido2.BlazorWebAssembly/Fido2.BlazorWebAssembly.csproj [hunk 1]
```diff
@@ -39,4 +39,8 @@
     </Content>
   </ItemGroup>
 
+  <ItemGroup>
+    <PackageReference Update="Microsoft.SourceLink.GitHub" Version="8.0.0" />
```
Comment (joegoldman2) [inline Src/Fido2.BlazorWebAssembly/Fido2.BlazorWebAssembly.csproj:43]: Do we need this package reference? Starting with .NET 8, Source Link is included in the .NET SDK and enabled by default: https://github.com/dotnet/sourcelink/?tab=readme-ov-file#using-source-link-in-net-projects

Comment (iamcarbon) [reply Src/Fido2.BlazorWebAssembly/Fido2.BlazorWebAssembly.csproj:43]: Probably not. Let me see if I can remove it.

File: Demo/Demo.csproj [hunk 2]
```diff
@@ -13,8 +13,8 @@
     <ProjectReference Include="..\Src\Fido2\Fido2.csproj" />
   </ItemGroup>
 
-  <ItemGroup Condition=" '$(TargetFramework)' == 'net6.0' ">
-    <PackageReference Include="Microsoft.VisualStudio.Web.CodeGeneration.Design" Version="6.0.0" />
+  <ItemGroup Condition=" '$(TargetFramework)' == 'net8.0' ">
```
Comment (joegoldman2) [inline Demo/Demo.csproj:16]: Is this condition necessary? It always seems to be evaluated to true.

File: azure-pipelines.yml [hunk 3]
```diff
@@ -31,17 +31,17 @@ jobs:
       arguments: '--configuration $(buildConfiguration) --no-restore --nologo "-p:Version=$(Build.BuildNumber)-development"'
       projects: 'Demo/Demo.csproj'
   - task: DotNetCoreCLI@2
-    displayName: Publish website [net6.0]
+    displayName: Publish website [net8.0]
     inputs:
       command: publish
-      arguments: '--no-build --nologo --configuration $(BuildConfiguration) --output $(build.ArtifactStagingDirectory)/net6.0 /p:PackageVersion=$(Build.BuildNumber)-development --framework net6.0'
+      arguments: '--no-build --nologo --configuration $(BuildConfiguration) --output $(build.ArtifactStagingDirectory)/net8.0 /p:PackageVersion=$(Build.BuildNumber)-development --framework net8.0'
```
Comment (joegoldman2) [inline azure-pipelines.yml:37]:
```suggestion
      arguments: '--no-build --nologo --configuration $(BuildConfiguration) --output $(build.ArtifactStagingDirectory)/net8.0 /p:PackageVersion=$(Build.BuildNumber)-development'
```

Do we need to specify the target framework here? The demo project only targets one version at a time.

File: Src/Fido2/Metadata/ConformanceMetadataRepository.cs [hunk 4]
```diff
@@ -56,7 +58,9 @@ public async Task<MetadataBLOBPayload> GetBLOBAsync(CancellationToken cancellati
 
         var content = new ByteArrayContent(JsonSerializer.SerializeToUtf8Bytes(req, FidoSerializerContext.Default.GetBLOBRequest))
         {
-            Headers = { { "Content-Type", "application/json" } }
+            Headers = {
```
Comment (joegoldman2) [inline Src/Fido2/Metadata/ConformanceMetadataRepository.cs:61]: nit:
```suggestion
            Headers =
            {
```

**Review by joegoldman2 — COMMENTED** [review]
I just left a few comments. Otherwise, this migration looks awesome!

File: Src/Fido2/Metadata/ConformanceMetadataRepository.cs (hunk 4 above)
Comment (iamcarbon) [reply Src/Fido2/Metadata/ConformanceMetadataRepository.cs]: updated.

File: Demo/Demo.csproj (hunk 2 above)
Comment (iamcarbon) [reply Demo/Demo.csproj]: removed.

**Comment by iamcarbon** [comment]
This also addresses [CVE-2024-21319](https://github.com/advisories/GHSA-8g9c-28fc-mcx2)

@abergs Let me know when you have a chance to review.

**Review by abergs — APPROVED** [review]
Looks great!

**Pull request merged**